definitions/npm/styled-components_v2.x.x/flow_v0.53.x-/styled-components_v2.x.x.js

```diff
@@ -0,0 +1,372 @@
+  polygon:                  $npm$styledComponents$StyledComponentsComponentListValue,
+  polyline:                 $npm$styledComponents$StyledComponentsComponentListValue,
+  radialGradient:           $npm$styledComponents$StyledComponentsComponentListValue,
+  rect:                     $npm$styledComponents$StyledComponentsComponentListValue,
+  stop:                     $npm$styledComponents$StyledComponentsComponentListValue,
+  svg:                      $npm$styledComponents$StyledComponentsComponentListValue,
+  text:                     $npm$styledComponents$StyledComponentsComponentListValue,
+  tspan:                    $npm$styledComponents$StyledComponentsComponentListValue,
+|}
+
+declare module 'styled-components' {
+  declare type Interpolation                                              = $npm$styledComponents$Interpolation;
+  declare type NameGenerator                                              = $npm$styledComponents$NameGenerator;
+  declare type Theme                                                      = $npm$styledComponents$Theme;
+  declare type ThemeProviderProps                                         = $npm$styledComponents$ThemeProviderProps;
+  declare type TaggedTemplateLiteral<R>                                   = $npm$styledComponents$TaggedTemplateLiteral<R>;
+  declare type ComponentListKeys                                          = $npm$styledComponents$StyledComponentsComponentListKeys;
+
+  declare type ReactComponentFunctional<Props: {}, DefaultProps: ?{} = *> = $npm$styledComponents$ReactComponentFunctional<Props, DefaultProps>;
+  declare type ReactComponentFunctionalUndefinedDefaultProps<Props: {}>   = $npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props>;
+  declare type ReactComponentClass<Props: {}, DefaultProps: ?{} = *>      = $npm$styledComponents$ReactComponentClass<Props, DefaultProps>;
+  declare type ReactComponentClassUndefinedDefaultProps<Props: {}>        = $npm$styledComponents$ReactComponentClassUndefinedDefaultProps<Props>;
+  declare type ReactComponentUnion<Props>                                 = $npm$styledComponents$ReactComponentUnion<Props>;
+  declare type ReactComponentIntersection<Props>                          = $npm$styledComponents$ReactComponentIntersection<Props>;
+  declare type ReactComponentStyledStaticProps<Props>                     = $npm$styledComponents$ReactComponentStyledStaticProps<Props, ComponentListKeys>;
+  declare type ReactComponentStyled<Props>                                = $npm$styledComponents$ReactComponentStyled<Props, ComponentListKeys>;
+  declare type ReactComponentStyledTaggedTemplateLiteral<Props>           = $npm$styledComponents$ReactComponentStyledTaggedTemplateLiteral<Props, ComponentListKeys>;
+
+  declare module.exports: {
+    <Props, Key: ComponentListKeys>(Key | ReactComponentUnion<Props>): ReactComponentStyledTaggedTemplateLiteral<Props>,
```

Comment:
Why take type argument `Key` instead of just taking `(ComponentListKeys | ReactComponentUnion<Props>)`?

Reply:
The funny bit is that I originally had it as ``(ComponentListKeys | ReactComponentUnion<Props>)``, but changed it because I wrongly _remembered_ that ``ComponentListKeys`` wasn't inside a ``$Subtype``.

Updated.

Reply:
I'm actually about to change it anyways to address this: https://github.com/flowtype/flow-typed/pull/1156#pullrequestreview-58414828